test/test_data/test_feature_validator.py

```diff
@@ -419,7 +418,9 @@ def test_list_to_dataframe(openml_id):
             # convert dtype translates 72.0 to 72. Be robust against this!
             assert is_numeric_dtype(transformed_X[i].dtype)
         else:
-            assert X_pandas[col].dtype.name == transformed_X[i].dtype.name, col
+            if X_pandas[col].dtype.name == "category":
```

Comment:
Do you want to treat category and string separately here? If yes, please put them into two elifs and use and extra else in which you raise a `TypeError` to catch other unwanted types.

Reply:
We should in general rethink the feature validator a bit. And maybe think about strategies to automaticlly label 'objects' as either stirngs or categories